hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/webapp/FederationInterceptorREST.java

```diff
@@ -1366,4 +1386,49 @@ public void shutdown() {
       threadpool.shutdown();
     }
   }
+
+  <R> Map<SubClusterInfo, R> invokeConcurrent(Collection<SubClusterInfo> clusterIds,
+       ClientMethod request, Class<R> clazz) throws YarnException, IOException {
+    ArrayList<SubClusterInfo> clusterIdList = new ArrayList<>(clusterIds);
+    return invokeConcurrent(clusterIdList, request, clazz);
+  }
+
+  private <R> Map<SubClusterInfo, R> invokeConcurrent(ArrayList<SubClusterInfo> subClusterInfo,
+      ClientMethod request, Class<R> clazz) {
+
+    Map<SubClusterInfo, R> results = new HashMap<>();
+
+    // Send the requests in parallel
+    CompletionService<R> compSvc = new ExecutorCompletionService<>(this.threadpool);
+
+    for (final SubClusterInfo info : subClusterInfo) {
```

Comment:
You could pass directly:
```
Collection<SubClusterInfo> clusterIds
```
No need to make it an array list.

Reply:
Thanks a lot for the advice, let me learn a lot, the code has been modified.